### ISSUE TITLE
HTTP PUT request will return 400 ERROR when rapidly switch shipping method in Delivery Mode step during checkout

### ISSUE THREAD

**Issue by wangzixi-diablo** [SAP/spartacus#14386]
**Describe the bug**

When customer switches shipping method rapidly in Delivery Mode maintenance step during checkout process, HTTP PUT request will receive 400 Error code returned from SAP Commerce Cloud backend.

**Tell us the version of Spartacus**
- Library version: 4.2.0

**To Reproduce**
1. launch SAP Spartacus locally
2. select one product, add to cart and proceed to checkout 
3. In Delivery Mode maintenance step, toggle between "Standard Delivery" and "Premium Delivery" rapidly
![image](https://user-images.githubusercontent.com/58975336/142798237-87663e8b-3062-452e-bfd4-49346797edfa.png)

4. Repeat the step until we see HTTP 400 error in Chrome dev tool console:

![image](https://user-images.githubusercontent.com/58975336/142798582-eb74b957-1fb6-4203-a9ce-226586e6f545.png)

There are two types of errors:

1. HTTP PUT request: https://\<host\>:9002/occ/v2/electronics-spa/users/current/carts/00001001/deliverymode?deliveryModeId=standard-gross&lang=en&curr=USD

error:
```json
{
   "errors" : [ {
      "message" : "The application has encountered an error",
      "type" : "JaloObjectNoLongerValidError"
   } ]
}
```
![image](https://user-images.githubusercontent.com/58975336/142798960-78fe38e3-05ef-4388-acfc-23cc1a3c2bd5.png)

2. HTTP GET request:

https://\<host\>:9002/occ/v2/electronics-spa/users/current/carts/00001001?fields=DEFAULT,potentialProductPromotions,appliedProductPromotions,potentialOrderPromotions,appliedOrderPromotions,entries(totalPrice(formattedValue),product(images(FULL),stock(FULL)),basePrice(formattedValue,value),updateable),totalPrice(formattedValue),totalItems,totalPriceWithTax(formattedValue),totalDiscounts(value,formattedValue),subTotal(formattedValue),deliveryItemsQuantity,deliveryCost(formattedValue),totalTax(formattedValue,%20value),pickupItemsQuantity,net,appliedVouchers,productDiscounts(formattedValue),user,saveTime,name,description&lang=en&curr=USD

error:

```json
{
   "errors" : [ {
      "message" : "The application has encountered an error",
      "type" : "JaloObjectNoLongerValidError"
   } ]
}
```
![image](https://user-images.githubusercontent.com/58975336/142798933-e2071541-2b9f-4acb-9027-eb16644d274a.png)

**Expected behavior**
No HTTP error even when customer switches shipping method in a very rapid way.

**Desktop (please complete the following information):**
 - OS: Windows 10 laptop
 - Browser: Chrome 95.0.4638.69

**Comment by wangzixi-diablo** [comment]
Every time you switch delivery modes, the cart will be recalculated in the backend: DefaultPromotionRuleActionService.recalculateTotals method in SAP Commerce Cloud backend is called.

Doing so requires deleting the promotion results on the cart (and the associated promotion actions) and calculating them from scratch with the new delivery mode in the backend.

When we do this fast enough in Spartacus UI, the backend code actually tries to retrieve the existing promotion results that were already deleted by the previous delivery mode HTTP PUT request, and get errors like below in backend:

```log
"ERROR","loggerName":"de.hybris.platform.webservicescommons.resolver.RestHandlerExceptionResolver","message":"item 8796519175063 no longer valid (was removed): object no longer valid
```

And then the backend exception is handled and corresponding error message is returned to Spartacus layer.

**Comment by Xymmer** [comment]
Workaround provided. Let's fix for 4.2.x and backport to 3.4.x.

**Comment by wangzixi-diablo** [comment]
In the fix of linked PR, we have decided to disable the delivery mode radio input control when there's already an on-going HTTP put request to change delivery mode in commerce cloud backend server. The reason is the spinner might give user an hint that there's some data currently being loaded from server, whereas in our case, there's no data being loaded, but the HTTP put request being executed.

Anyway based on feedback from Brian and Nik, I will use the spinner to "hide" input control to conform with all the spinners we have in checkout. Please track the change in https://github.com/SAP/spartacus/issues/14710
